web-common/src/features/sources/modal/SourceImportedModal.svelte

```diff
@@ -24,6 +27,7 @@
   }
 
   async function autoGenerateDashboard() {
+    if (dashboardFromSourceUIAction === null) return;
```

Comment:
Another case – can we instead disable the Button?

Reply:
disabled the button, but we still need this type check, because the function scope doesn't know about the temporal order of the events. added a comment explaining.